solidity/src/Router7683.sol

```diff
@@ -61,4 +79,13 @@ contract Router7683 is GasRouter, Base7683 {
     function _localDomain() internal view override returns (uint32) {
         return localDomain;
     }
+
+    function _mustHaveSameOrigin(bytes32[] memory _orderIds) internal view returns (uint32 originDomain) {
```

Comment:
Is there a way we can avoid this gas expenditure? I.e. if an order that has a different origin domain is included in the error, can we just ignore on the origin chain? I'd assume that solvers will care about gas consumption

Reply:
I think we could, but we need a way to determine where to send the message and I don't think we can get this from the orders. Maybe adding an explicit param to the `Base7386.settle` function?

Reply:
I prefer not to add the para to the Base7386  `settle` and let the inheriting contract to handle it. Maybe some one wants to handle multi destinations settling.

Reply:
imo, its fine to just pick the first one, i think pushing this concern into the settlement-specific off-chain logic makes sense to me. Agree to not add this to base settle